tools/clang/include/clang/Sema/Sema.h

```diff
@@ -3807,6 +3808,9 @@ class Sema {
                                         QualType TargetType,
                                         SourceLocation Loc);
   bool DiagnoseHLSLMethodCall(const CXXMethodDecl *MD, SourceLocation Loc);
+  void DiagnoseUsedHLSLMethodCall(const CXXMethodDecl *MD, SourceLocation Loc,
```

Comment:
nit: this method name doesn't really convey what you're diagnosing. This isn't diagnosing that a method call is unused, but rather diagnosing _availability_ of methods regardless of liveness.

Reply:
Only method call which could reached from entry or exported function will be diagnosed.
How about DiagnoseReachableHLSLMethodCall?

Reply:
@python3kgae, were you going to update this name?

Reply:
Fixed.